src/main/java/org/triplea/ui/TripleA.java

```diff
@@ -0,0 +1,104 @@
+package org.triplea.ui;
+
+import java.net.URL;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.Image;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class TripleA extends Application {
+
+  @FXML
+  private VBox loadingOverlay;
+
+  @FXML
+  private VBox exitOverlay;
+
+  @FXML
+  private Tooltip progressTooltip;
+
+  @FXML
+  private Label progressLabel;
+
+  @FXML
+  private StackPane rootPane;
+
+  private TripleaMainMenu mainMenu;
+
+  @Override
+  public void start(final Stage stage) throws Exception {
+    final FXMLLoader loader = getLoader(getClass().getResource("./fxml/TripleAMain.fxml"));
+    loader.setController(this);
+    final Scene scene = new Scene(loader.load(), 960, 540);
+    scene.getStylesheets().add("org/triplea/ui/css/main.css");
+    final TripleaMainMenu mainMenu = addRootContent(new TripleaMainMenu(this));
+    this.mainMenu = mainMenu;
+    stage.setMinHeight(scene.getHeight());
+    stage.setMinWidth(scene.getWidth());
+    stage.setScene(scene);
+    stage.getIcons().add(new Image(getClass().getResourceAsStream("/games/strategy/engine/framework/ta_icon.png")));
+    stage.setTitle("TripleA");
+    stage.show();
+    stage.setOnCloseRequest(e -> System.exit(0));
+  }
+
+  public static void launch(final String... arg) {
+    Application.launch(arg);
+  }
+
+  /**
+   * Simplified way of getting an {@link FXMLLoader} with the default settings for TripleA.
+   * @param location The FXML File to load
+   * @return An FXMLLoader object
+   */
+  public static FXMLLoader getLoader(final URL location) {
+    final FXMLLoader loader = new FXMLLoader();
+    loader.setLocation(location);
+    loader.setResources(ResourceBundle.getBundle("org.triplea.ui.lang.TripleA", new Locale("en", "US")));
+    return loader;
+  }
+
+  public TripleaMainMenu getMainMenu() {
+    return mainMenu;
+  }
+
+  public void promptExit() {
+    exitOverlay.setVisible(true);
+  }
+
+  <T extends Node> T addRootContent(final T node) {
+    rootPane.getChildren().add(node);
+    return node;
+  }
+
+  @FXML
+  private void hideExitConfirm() {
+    exitOverlay.setVisible(false);
+  }
+
+  @FXML
+  private void exit() {
+    Platform.exit();
+    System.exit(0);
+  }
+
+  public void displayLoadingScreen(final boolean bool) {
```

Comment:
Nothing calls this yet?
I think we may want to discuss a bit how we are going to control the GUI. I am reminded of some of the existing problems  in the current GUI classes. For example here we have some other class that is really aware of the full blown application state and when the loading screen should and should not be shown.

To continue the thought, I've been thinking we may want to introduce the concept of UI events and have a UI event stack somewhere. Based on the events in this stack, probably only needing the last one or two, we can determine the application UI state. Then we would tie (idempotent) behavior to entering those states.

I don't know if it would be good to create a whole new control layer into a new UI layer. We would be getting close to the re-write the app path (and TripleA is a large code base, it is a no-go).

This does start to open up a yet larger conversation. It might be worthwhile for us to start examining very closely how the UI is tied to state and game engine logic. If we can come up with a nicer event system, it may aid us in decoupling the UI. Then the transition to JavaFx woudl be much more 1:1. Otherwise I'm concerned we are very coupled to the existing code structure while this migration happens. First if we overhaul the UI, this is out of sync.. Second, since we don't have a good architecture for the UI, we see some early problems that are starting to bake. Given we are replacing UI and can't do too much more, we would inherit much of the existing structural problems from the original UI code.

Note: I'm just a bit concerned we're going to be backed in a corner. I have thought we should focus a bit first on decoupling UI and logic in the existing code to make migration easier. Regardless, I do think it is pretty essential that we step back a bit and spend a few days or a week or two to design a better control/architecture for the UI.  Thinking about that, it might be a good jumping board for splitting the UI code quickly, and then  this migration work would should go much faster (since you woudl not have to re-invent how for example UI state is controlled, we would have solved that already somewhere else once for the entire system)